python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -1142,6 +1160,31 @@ def submit_run_request(
                     )
                     updated_evaluation_keys.add(entity_key)
 
+            # check if the sensor is still enabled:
+            if num_submitted >= batch_size:
+                num_submitted = 0
+                should_terminate = False
+                use_auto_materialize_sensors = instance.auto_materialize_use_sensors
+                if not use_auto_materialize_sensors and get_auto_materialize_paused(instance):
```

Comment:
maybe if (not use_auto_materialize_sensors) and get_auto_materialize_paused(instance): - to make sure the ordering is explicit